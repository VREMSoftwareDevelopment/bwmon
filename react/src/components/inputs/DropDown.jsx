/*
 *      Copyright (C) 2010 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Select, MenuItem } from '@mui/material';

const DropDown = ({ items, ...props }) => {
    return items && props.value ? (
        <Select {...props}>
            {items.map((item, index) => (
                <MenuItem key={index} value={item}>
                    {item}
                </MenuItem>
            ))}
        </Select>
    ) : null;
};

// Stryker disable next-line all
DropDown.propTypes = { items: PropTypes.array, value: PropTypes.any };

export default DropDown;
