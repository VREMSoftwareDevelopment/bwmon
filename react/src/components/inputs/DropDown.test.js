/*
 *      Copyright (C) 2010 - 2023 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { create } from 'react-test-renderer';
import DropDown from './DropDown';

describe('DropDown', () => {
    const handleChange = (event) => {
        // handle change
    };

    it('renders correctly with items', () => {
        const tree = create(
            <DropDown id="12345" value="ZYXUW" onChange={handleChange} items={['ABC', 'ZYXUW', 'WGKF']} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with no items', () => {
        const tree = create(<DropDown />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with no value', () => {
        const tree = create(<DropDown items={['ABC', 'ZYXUW', 'WGKF']} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
