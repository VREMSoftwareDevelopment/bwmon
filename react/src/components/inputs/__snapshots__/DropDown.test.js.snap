// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown renders correctly with items 1`] = `
<div
  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  onClick={[Function]}
>
  <div
    aria-controls=":r0:"
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="12345"
    className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
    id="12345"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
    tabIndex={0}
  >
    ZYXUW
  </div>
  <input
    aria-hidden={true}
    aria-invalid={false}
    className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
    value="ZYXUW"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
    data-testid="ArrowDropDownIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
  <fieldset
    aria-hidden={true}
    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
  >
    <legend
      className="css-ihdtdm"
    >
      <span
        className="notranslate"
      >
        â€‹
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`DropDown renders correctly with no items 1`] = `null`;

exports[`DropDown renders correctly with no value 1`] = `null`;
